### ISSUE TITLE
NPE: Cannot invoke "String.getBytes()" because "key" is null

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/zeebe#9269]
**Describe the bug**

Evaluating a decision results in a NullPointerException. The NPE occurs during evaluation of a decision. It appears to be iterating over a map of variables and breaks when it encounters a variable with the key `null`.

**To Reproduce**

Not sure

**Expected behavior**

No NPE should occur. If any null values are possible we should be able to deal with them.

**Log/Stacktrace**

https://console.cloud.google.com/errors/detail/CP_pnKWJn4Ta9AE;service=zeebe;time=P7D?project=camunda-cloud-240911

<details><summary>Full Stacktrace</summary>
 <p>

```
 java.lang.NullPointerException: Cannot invoke "String.getBytes()" because "key" is null
	at io.camunda.zeebe.feel.impl.FeelToMessagePackTransformer.$anonfun$writeValue$2(FeelToMessagePackTransformer.scala:50) ~[zeebe-feel-integration-8.0.0.jar:8.0.0]
	at scala.collection.immutable.Map$Map2.foreach(Map.scala:342) ~[scala-library-2.13.8.jar:?]
	at io.camunda.zeebe.feel.impl.FeelToMessagePackTransformer.writeValue(FeelToMessagePackTransformer.scala:49) ~[zeebe-feel-integration-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.feel.impl.FeelToMessagePackTransformer.$anonfun$writeValue$1(FeelToMessagePackTransformer.scala:42) ~[zeebe-feel-integration-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.feel.impl.FeelToMessagePackTransformer.$anonfun$writeValue$1$adapted(FeelToMessagePackTransformer.scala:42) ~[zeebe-feel-integration-8.0.0.jar:8.0.0]
	at scala.collection.immutable.List.foreach(List.scala:333) ~[scala-library-2.13.8.jar:?]
	at io.camunda.zeebe.feel.impl.FeelToMessagePackTransformer.writeValue(FeelToMessagePackTransformer.scala:42) ~[zeebe-feel-integration-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.feel.impl.FeelToMessagePackTransformer.toMessagePack(FeelToMessagePackTransformer.scala:27) ~[zeebe-feel-integration-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.dmn.impl.DmnScalaDecisionEngine.toMessagePack(DmnScalaDecisionEngine.java:170) ~[zeebe-dmn-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.dmn.impl.EvaluatedDmnScalaDecision.of(EvaluatedDmnScalaDecision.java:56) ~[zeebe-dmn-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.dmn.impl.DmnScalaDecisionEngine.lambda$getEvaluatedDecisions$1(DmnScalaDecisionEngine.java:162) ~[zeebe-dmn-8.0.0.jar:8.0.0]
	at scala.collection.immutable.List.foreach(List.scala:333) ~[scala-library-2.13.8.jar:?]
	at io.camunda.zeebe.dmn.impl.DmnScalaDecisionEngine.getEvaluatedDecisions(DmnScalaDecisionEngine.java:159) ~[zeebe-dmn-8.0.0.jar:8.0.0]
	at java.util.Optional.map(Unknown Source) ~[?:?]
	at io.camunda.zeebe.dmn.impl.DmnScalaDecisionEngine.evaluateDecisionById(DmnScalaDecisionEngine.java:105) ~[zeebe-dmn-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnDecisionBehavior.evaluateDecisionInDrg(BpmnDecisionBehavior.java:182) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnDecisionBehavior.lambda$evaluateDecision$3(BpmnDecisionBehavior.java:111) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.Either$Right.flatMap(Either.java:366) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnDecisionBehavior.evaluateDecision(BpmnDecisionBehavior.java:109) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.task.BusinessRuleTaskProcessor$CalledDecisionBehavior.lambda$onActivate$0(BusinessRuleTaskProcessor.java:89) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.Either$Right.flatMap(Either.java:366) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.task.BusinessRuleTaskProcessor$CalledDecisionBehavior.onActivate(BusinessRuleTaskProcessor.java:89) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.task.BusinessRuleTaskProcessor.onActivate(BusinessRuleTaskProcessor.java:40) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.task.BusinessRuleTaskProcessor.onActivate(BusinessRuleTaskProcessor.java:21) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processEvent$2(BpmnStreamProcessor.java:128) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:381) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:127) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processRecord$0(BpmnStreamProcessor.java:110) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:381) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:107) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:300) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84) ~[zeebe-db-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processInTransaction(ProcessingStateMachine.java:290) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processCommand(ProcessingStateMachine.java:253) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextRecord(ProcessingStateMachine.java:213) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextRecord(ProcessingStateMachine.java:189) ~[zeebe-workflow-engine-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:79) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:44) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:97) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:80) ~[zeebe-util-8.0.0.jar:8.0.0]
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:189) ~[zeebe-util-8.0.0.jar:8.0.0] 
```

</p>
</details>

**Environment:**
- OS: Camunda Cloud
- Zeebe Version: Seen in 8.0.0 and 8.0.1

**Comment by remcowesterhoud** [comment]
@saig0 You might be interested in this one. I haven't been able to fully analyse where the problem was coming from.

**Comment by saig0** [comment]
@remcowesterhoud I'm also not sure how to reproduce it :sweat_smile: The issue seems to be in the DMN engine.

It would be good to see the DMN that was evaluated. Can we get the DMN from the cluster for root cause analysis?

**Comment by npepinpe** [comment]
@remcowesterhoud please try to get the the DMN to further analyze this.

The expected impact here is that the instance is blacklisted, so not visible to the user, but the cluster will keep "working". We also expect likelihood here to be low, as no one ran into this so far. I'll mark it as blocker/info and severity/low for now, but tentatively, and once we have the DMN we should revise this.

**Comment by remcowesterhoud** [comment]
I have identified and reproduces the issue. See https://github.com/camunda-community-hub/dmn-scala/issues/142

**Comment by remcowesterhoud** [comment]
Will be fixed by #9361
